app/backend/src/couchers/jobs/handlers.py

```diff
@@ -131,6 +137,43 @@ def process_generate_message_notifications(payload):
             )
 
 
+def process_event_creation_emails(payload):
+    """Sends out emails to all users subscribed to a newly created event's cluster."""
+    logger.info(f"Sending out emails for event_id = {payload.event_id}")
+
+    with session_scope() as session:
+        event = session.execute(select(Event).where(Event.id == payload.event_id)).one()
+        creator = event.creator_user
+
+        # can be replaced by the frontend preventing an incomplete profile from creating an event
+        if not creator.has_completed_profile():
+            logger.info(f"{creator.name=} can't send emails for {event.name=} because of an incomplete profile.")
+            return
+
+        subscribers: set[User] = set()
+        for cluster in event.clusters:
+            if cluster.parent_node_id == 1:
+                logger.info(f"The Global Community is too big for email notifications.")
+            elif cluster.is_leaf or creator in cluster.admins:
+                subscribers.update(cluster.members)
+            else:
+                max_radius = 20000
+                nearby_subscribers = session.execute(
+                    select(User)
+                    .join(ClusterSubscription, ClusterSubscription.user_id == User.id)
+                    .where(ClusterSubscription.cluster_id == cluster.id)
+                    .where(func.ST_Contains(func.ST_Buffer(event.occurrences[0].geom, max_radius), User.geom))
```

Comment:
I think `func.ST_DWithin(User.geom, event.occurrences[0].geom, max_radius / 111111)` is preferred, according to the web. These geography fields use WGS84 coordinates, so distances are measured in degrees; a degree is ~111km at the equator.